Analysis #69 DC vacant lot analysis

maggiesu0725 opened the pull request.

maggiesu0725 committed:
initial set up

maggiesu0725 committed:
categorize dwelling type

maggiesu0725 committed:
recode dwelling type

maggiesu0725 committed:
clean vacant lot data

maggiesu0725 committed:
replace zoning code with shapefile zoning code

maggiesu0725 committed:
clean zoning code inconsistency

maggiesu0725 committed:
first run of code

maggiesu0725 committed:
more clean up of residential code

maggiesu0725 committed:
investigate lotsize_sf less than lot_area_min issue

maggiesu0725 committed:
comment out parts not knitting. reproject parcel base geo for spatial joining

maggiesu0725 commented:
@lhendey I'm still working on reprojecting the parcel base geography for spatial joining instead of using the blacknight lat and lon. But since DC blacknight data zoning information is outdated, the program is a bit different than Sarah's example. I thought it would be helpful for you to just take a look at the structure and procedure so I can adjust accordingly while figuring out some programming issues. Thanks!

maggiesu0725 commented:
@sstrochak will you have time to take a quick look at the DC version vacant lot?

sstrochak commented:
@maggiesu0725 yeah, is it ready now?

maggiesu0725 commented:
I haven't solved the projection issue but just want to get some general feedback on quality checks. DC blacknight zoning is outdated so the part where you compared the blacknight and shapefile zoning is not applicable in this case, I also commented in the program where the original methodology is not applicable. If you could take a look at those. Thanks! @sstrochak

sstrochak commented:
@maggiesu0725 I can get to it either tomorrow morning or Wednesday morning, does that work?

maggiesu0725 commented:
Sure take your time! or if you prefer I walk you through the code sometime tomorrow. @sstrochak

maggiesu0725 committed:
reproject parcelgeo xy

maggiesu0725 committed:
export csv for arcgis

maggiesu0725 committed:
furthur breakdown factor variable, calculate FAR units, update units flag

maggiesu0725 committed:
update unit total

maggiesu0725 committed:
remove NA in calculation

maggiesu0725 committed:
subdivide karge vacant lots that have dwelling unit limits

maggiesu0725 commented:
So I've subdivided the lots with #dwelling restrictions. Could you do a final review? @sstrochak @lhendey

maggiesu0725 committed:
update newdwellingtype categorization, add more explanation throughout program, adjust order of program

maggiesu0725 commented:
@lhendey I made all the changes and added the number of observation I got in the comment so it will be easier for you to compare,

lhendey committed:
leah's html

lhendey commented:
@maggiesu0725 - my html file after knitting still has different #s. if you knit the markdown to html do you get the same result as me?

lhendey commented on Prog/vacant-lots-DC.Rmd:
```diff
@@ -0,0 +1,886 @@
+
+## Calculate missing units
+
+There are several variables we will use to calculate how many units we can put on each vacant lot.
```
edit this comment to reflect dc calculations

lhendey reviewed (changes requested):
@maggiesu0725 see comments. 

cc @sstrochak .

maggiesu0725 committed:
update zoning code

maggiesu0725 committed:
remove 1 floor from mixed use development

maggiesu0725 committed:
recategorize dwelling type

maggiesu0725 commented:
@lhendey  ready for review

lhendey commented on Prog/vacant-lots-DC.Rmd:
````diff
@@ -0,0 +1,886 @@
+```
+
+There are 2228 R1s that are not joined which is a lot, in our spreadsheet we have R1A and R1B, pull in lotsize and see which ones we should assign the R1 to. R1A has lot size of 7500, R1B has lot size of 5000. Use as threshold.
+
````
@maggiesu0725 is this issue resolved?

lhendey committed:
corrected living area for FAR

lhendey commented:
@maggiesu0725 - i corrected the living area variable in the units_far calc and it broke the scatter plot function...

is there anyway to save the result of the get_unit_summary - i'd like to look at a few of the obs  where units_Far is getting us a lot of units to figure out whether the calc is wrong.

pspauster commented on Prog/vacant-lots-DC.Rmd:
```diff
@@ -0,0 +1,886 @@
+           units_per_floor = floor(living_area / average_unit_size),
+           floors = floor(height_ft / 10),
+           units_height = units_per_floor * floors,
+           units_FAR= (living_area*floor_area_ratio_max)/average_unit_size,
```
@lhendey do you know how we would adapt this calculation if there is no lot coverage specified?

maggiesu0725 commented:
I added the dataset **calculateunit_data**  which should be what you are looking for. It is in the calculation section. fixed on function and reknitted the document, should not be breaking now. @lhendey

maggiesu0725 committed:
export dataset for fact checking, fix function

maggiesu0725 commented:
also exported the dataset to CSV: L:/Libraries/RegHsg/Prog/Tables/DC-vacant-lots-analysis-basedata.csv @lhendey

maggiesu0725 committed:
export dataset

lhendey commented:
@maggiesu0725 thanks that's helpful. can you add the ssl to output?

Also - can you add units_far should be NA if floor_area_ratio_max is NA? (the zero's are trumping the units_height right now.

maggiesu0725 committed:
update Far units calculation, add parcel id to output table

maggiesu0725 commented:
thanks, changes made. @lhendey

lhendey committed:
changed all 900s to 940

maggiesu0725 committed:
calculate units for all lots, update FAR units calculation.

maggiesu0725 committed:
finalize and knit

maggiesu0725 committed:
update slicing rule

maggiesu0725 committed:
add comments for large SF lots

maggiesu0725 commented:
@lhendey I just knitted the document, if you want to take a look.

maggiesu0725 committed:
Merge branch 'master' into analysis-#69

maggiesu0725 committed:
write out r dataset

maggiesu0725 committed:
incorportate leah's comments, investigate ties in category

maggiesu0725 committed:
fix ties in category

maggiesu0725 committed:
minor bugfix

maggiesu0725 committed:
knit updated dc

lhendey reviewed (commented):
@maggiesu0725 

line 571 shouldn't we set jurtotal instead of jur2 to create jurcleaned?  they have different numbers of obs. 

line 621 jurzoningdata1 has less obs than jurtotal and jurcleaned. what is getting lost? can you add a comment?

lhendey committed:
updated 1 comment

lhendey committed:
Merge branch 'analysis-#69' of https://github.com/NeighborhoodInfoDC/RegHsg into analysis-#69

lhendey committed:
removed old code

lhendey commented:
@maggiesu0725 just pushed a few minor changes.

maggiesu0725 commented:
@lhendey  I added these comments int eh program as well.
line 571 shouldn't we set jurtotal instead of jur2 to create jurcleaned? they have different numbers of obs.
No we are using jur2 because jur2 has only the observations that have either parcelbase geo or blacknight coordinates and joined with the zoning shapefile to get the zoning information. 

line 621 jurzoningdata1 has less obs than jurtotal and jurcleaned. what is getting lost? can you add a comment?
I'm filtering out observations with zoning code that are not in the DC spreadsheet, because they are not going to be calculated in the next step. I might need to take the filter out.

maggiesu0725 committed:
add more comment on the code

maggiesu0725 committed:
add explanation on what codes got filtered out

lhendey committed:
add phaseland to preclean

lhendey committed:
add parcel_base land and improvement values

lhendey commented:
@maggiesu0725 -FYI i'm working on the vacant-dc file

lhendey committed:
changes to clean dc to add phasebuild

lhendey committed:
correcting vacant category with phasebuild

lhendey committed:
Update clean-DC.html

sstrochak commented:
To reclassify the addresses:
```
#create jur4 from blackknight address
jur5 <- jur4 %>% 
  filter(category_detail!="Garage") %>%
  classify_parceladdresses() %>% 
  mutate(ID = row_number()) %>% 
  mutate(address_type = ifelse(propaddress %in% c("address", "address", "address"),
                                                 "single",
                                                  address_type))
```

maggiesu0725 committed:
update far calculation

maggiesu0725 commented:
> To reclassify the addresses:
> 
> ```
> #create jur4 from blackknight address
> jur5 <- jur4 %>% 
>   filter(category_detail!="Garage") %>%
>   classify_parceladdresses() %>% 
>   mutate(ID = row_number()) %>% 
>   mutate(address_type = ifelse(propaddress %in% c("address", "address", "address"),
>                                                  "single",
>                                                   address_type))
> ```

@sstrochak what's the context of this? Is this something I need to update in the program? @lhendey

lhendey commented:
@maggiesu0725 - this is something i'm working on changing in the clean-dc file to get rid of lots that were incorrectly collapsing  and creating those huge # of units. cc @sstrochak

lhendey committed:
add code to reclassify nonres to single except for office condo/retail condo

lhendey committed:
Merge branch 'analysis-#69' of https://github.com/NeighborhoodInfoDC/RegHsg into analysis-#69

lhendey committed:
final clean up on clean-dc & knitted file

maggiesu0725 committed:
update DC far calculation, remove mixed use story calculation, knit document

maggiesu0725 commented:
@lhendey ready for review.

lhendey committed:
add mix1txtype to id cemeteries

lhendey committed:
add more exclusions to vacant lots

lhendey committed:
update preclean

maggiesu0725 committed:
update clean DC, vacant lot dc for parks, cemetry and educational

lhendey committed:
fixes to vacant lots to correct dataset

maggiesu0725 committed:
udpate education and cemetery filter

lhendey committed:
Update vacant-lots-DC.html

lhendey committed:
add new_land to preclean_dc

lhendey committed:
add new_impr

lhendey committed:
add new_impr to clean-dc

lhendey committed:
Update vacant-lots-DC.Rmd

lhendey committed:
add new_impr and update lot coverage max if na

lhendey committed:
add units_total to output for basedata
